tests/testthat/test-rng.R

```diff
@@ -515,6 +515,174 @@ test_that("exponential random numbers from floats have correct distribution", {
 })
 
 
+test_that("multinomial algorithm is correct", {
+  set.seed(1)
+  prob <- runif(10)
+  prob <- prob / sum(prob)
+  size <- 20
+  n <- 5
+
+  res <- dust_rng$new(1, seed = 1L)$multinomial(n, size, prob)
+
+  ## Separate implementation of the core algorithm:
```

Comment:
Also compare with `rmultinom()` results?

Reply:
I can't remember how the other distributions are tested, whether they're still compared to the R implementations, or if it's all done via mean and stddev now

Reply:
all against expectations now